Lib/test/test_named_expressions.py

```diff
@@ -513,6 +513,16 @@ def g():
             self.assertEqual(nonlocal_var, None)
         f()
 
+    def test_named_expression_scope_in_genexp(self):
+        a = 1
+        b = [1, 2, 3, 4]
+        genexp = (c := i + a for i in b)
+
+        with self.assertRaises(NameError):
+            print(c)  # c is not bound here
```

Comment:
Alternatively: `assert "c" not in locals()`

Reply:
I like this better, yeah.